gene2phenotype_project/gene2phenotype_app/serializers.py

```diff
@@ -112,49 +121,43 @@ def get_curators(self, id):
         return users
 
     def get_last_updated(self, id):
-        dates = []
-        lgd_panels = LGDPanel.objects.filter(panel=id)
-        for lgd_panel in lgd_panels:
-            if lgd_panel.lgd.date_review is not None and lgd_panel.lgd.is_reviewed == 1 and lgd_panel.lgd.is_deleted == 0:
-                dates.append(lgd_panel.lgd.date_review)
-                dates.sort()
-        if len(dates) > 0:
-            return dates[-1].date()
-        else:
-            return []
+        lgd_panel = LGDPanel.objects.filter(
+            panel=id,
+            lgd__is_reviewed=1,
+            lgd__is_deleted=0,
+            lgd__date_review__isnull=False
+            ).select_related('lgd'
+                             ).latest('lgd__date_review'
+                                      ).lgd.date_review
+
+        return lgd_panel.date() if lgd_panel else []
 
     # Calculates the stats on the fly
     # Returns a JSON object
     def calculate_stats(self, panel):
         lgd_panels = LGDPanel.objects.filter(
             panel=panel.id,
             is_deleted=0
-        ).select_related()
+        ).select_related('lgd__locus__type', 'lgd__disease')
 
         genes = set()
         diseases = set()
-        confidences = {}
         attrib_id = Attrib.objects.get(value='gene').id
+
         for lgd_panel in lgd_panels:
             if lgd_panel.lgd.locus.type.id == attrib_id:
                 genes.add(lgd_panel.lgd.locus.name)
 
             diseases.add(lgd_panel.lgd.disease_id)
 
-            try:
-                confidences[lgd_panel.lgd.confidence.value] += 1
-            except KeyError:
-                confidences[lgd_panel.lgd.confidence.value] = 1
-
         return {
             'total_records': len(lgd_panels),
             'total_genes': len(genes),
-            'total_diseases':len(diseases),
-            'by_confidence': confidences
```

Comment:
this stat is required for the panel page of the website

Reply:
Fixed